### DISCUSSION TITLE
Cannot subscribe to existing queue with no configure permission

### DISCUSSION THREAD

**Discussion by JOravas** [ruby-amqp/bunny#614] [General]
RabbitMQ version 3.8.11
Bunny 2.17.0

Preconditions:
- a permanent queue has been created in RabbitMQ outside of Bunny
- there is specific user for this queue that has only read/write access (no configure)

I cannot subscribe to this existing named queue from Bunny as there is ACCESS_REFUSED exception.

In RabbitMQ log:

operation queue.declare caused a channel exception access_refused.

**Comment by michaelklishin** [comment]
If your code involves declaring a queue — and `Bunny::Channel#queue` does just that — the user must have a `configure` permission on the target queue.

You can do a passive declare (pass `passive: true` to `Bunny::Channel#queue`) or use `Bunny::Channel#basic_consume` directly without all the DSL-y sugar.

**Reply by JOravas** [reply]
Thanks @michaelklishin. Code currently calls method **Channel#queue** and the documentation didn't have option _passive_ listed for it: http://reference.rubybunny.info/Bunny/Channel.html. I didn't either notice that for _queue_declare_ it is there and thus should have tried it. Problem solved with passive declare.

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(
